projects/extension/sql/ai--0.6.0.sql

```diff
@@ -0,0 +1,403 @@
+
+    -- if ai.indexing_default, resolve the default
+    if indexing operator(pg_catalog.->>) 'implementation' = 'default' then
+        indexing = ai._resolve_indexing_default();
+    end if;
+
+    -- validate the indexing config
+    perform ai._validate_indexing(indexing);
+
+    -- validate the formatting config
+    perform ai._validate_formatting(formatting, _source_schema, _source_table);
+
+    -- if ai.scheduling_default, resolve the default
+    if scheduling operator(pg_catalog.->>) 'implementation' = 'default' then
+        scheduling = ai._resolve_scheduling_default();
+    end if;
+
+    -- validate the scheduling config
+    perform ai._validate_scheduling(scheduling);
+
+    -- validate the processing config
+    perform ai._validate_processing(processing);
+
+    -- if scheduling is none then indexing must also be none
+    if scheduling operator(pg_catalog.->>) 'implementation' = 'none'
+    and indexing operator(pg_catalog.->>) 'implementation' != 'none' then
+        raise exception 'automatic indexing is not supported without scheduling. set indexing=>ai.indexing_none() when scheduling=>ai.scheduling_none()';
+    end if;
+
+    -- grant select to source table
+    perform ai._vectorizer_grant_to_source
+    ( _source_schema
+    , _source_table
+    , grant_to
+    );
+
+    -- create the target table
+    perform ai._vectorizer_create_target_table
+    ( _source_schema
+    , _source_table
+    , _source_pk
+    , target_schema
+    , target_table
+    , _dimensions
+    , grant_to
+    );
+
+    -- create queue table
+    perform ai._vectorizer_create_queue_table
+    ( queue_schema
+    , queue_table
+    , _source_pk
+    , grant_to
+    );
+
+    -- create trigger on source table to populate queue
+    perform ai._vectorizer_create_source_trigger
+    ( _trigger_name
+    , queue_schema
+    , queue_table
+    , _source_schema
+    , _source_table
+    , _source_pk
+    );
+
+    -- create view
+    perform ai._vectorizer_create_view
+    ( view_schema
+    , view_name
+    , _source_schema
+    , _source_table
+    , _source_pk
+    , target_schema
+    , target_table
+    , grant_to
+    );
+
+    -- schedule the async ext job
+    select ai._vectorizer_schedule_job
+    (_vectorizer_id
+    , scheduling
+    ) into _job_id
+    ;
+    if _job_id is not null then
+        scheduling = pg_catalog.jsonb_insert(scheduling, array['job_id'], to_jsonb(_job_id));
+    end if;
+
+    embedding_batch_schema = coalesce(embedding_batch_schema, 'ai');
+    embedding_batch_table = coalesce(embedding_batch_table, pg_catalog.concat('_vectorizer_embedding_batches_', _vectorizer_id));
+    embedding_batch_chunks_table = coalesce(embedding_batch_chunks_table, pg_catalog.concat('_vectorizer_embedding_batch_chunks_', _vectorizer_id));
+
+    -- create batch embedding tables
+    select (embedding operator (pg_catalog.->> 'implementation'))::text into _implementation;
+    if _implementation = 'openai' then
```

Comment:
This needs to be created only if the embedding implementation the user set is for openai batch. We have, I think 2 options. Add a new boolean attribute to the current `ai.embedding_openai` like `batch_api=true`, or `use_batch_api=true`. The other is to create a new embedding configuration for openai batch api.

The benefit I see of creating a new configuration, is that it'll be easier to handle the different logics. Also, both APIs don't support the exact same options, for example, the batch API doesn't support the `user` option and the non-batch doesn't require a `completion_window`.

You can create a new implementation and can call it `ai.embedding_openai_batch`. You can use the `ai.embedding_openai` one for reference:

https://github.com/timescale/pgai/blob/59c32947972bea851193a6a382006c169eed1969/projects/extension/sql/idempotent/008-embedding.sql?5C#L3-L22

You can create it in the same file.